### DISCUSSION TITLE
2.08 not reusing old masters

### DISCUSSION THREAD

**Discussion by AnttiKarvinen** [vitobotta/hetzner-k3s#477] [Q&A]
Tiny problem with Hetzner-k3s as i'm trying to update master nodes in out cloud (not in a rush or anything)
Our masters are named after the first  type of servers, data-cloud-cx21-master1
So i should use 
'set include_instance_type_in_instance_name:  true' 
-it is in my config but the master node type is now cx22 and new masters are made 
```
[Placement groups] Creating placement group data-cloud-nbg1-jupyterspark-4...
[Placement groups] Creating placement group data-cloud-nbg1-cx21-4...
[Placement groups] ...placement group data-cloud-nbg1-cx21-4 created
[Placement groups] ...placement group data-cloud-nbg1-jupyterspark-4 created
[Instance data-cloud-master2] Creating instance data-cloud-master2 (attempt 1)...
[Instance data-cloud-master3] Creating instance data-cloud-master3 (attempt 1)...
[Instance data-cloud-master1] Creating instance data-cloud-master1 (attempt 1)...
```
So how to reuse old masters, that are named about a year ago?

**Comment by vitobotta** [comment]
Hi, did you just change the instance type in the config file? That's not supported in clusters created with 1.x and will result in a new control plane being created, basically breaking the cluster. The `include_instance_type_in_instance_name` was added to allow the transition for older clusters, and 2.x no longer includes the instance type in the instance name, allowing you to change the instance type more easily without affecting existing nodes. But this is not possible with old clusters. I recommend you restore the previous instance type in the config file and re-run the create command to re-link the worker nodes to the previous control plane and restore the previous configuration. Then if you want to change the instance type of the old master, you can just use the rescale feature in the Hetzner Control Panel directly. The important bit is not to change the instance type in the config file because since it's included in the instance name with older clusters, it will create another control plane.

Before I can give you more instructions, what is the current state of the cluster?

The upcoming new release will include some improvements that make the instance type in the config file affect only new instances, to make changing instance types easier.

**Reply by AnttiKarvinen** [reply]
Great morning to Espoo, our cluster is working like it should,.
I deleted wrongly named cluster masters, so nothing unusual is in the cluster,.

I'll put our config here

<details><summary>Details</summary>
<p>


---
hetzner_token: 
cluster_name: data-cloud
kubeconfig_path: "./datakube"
k3s_version: v1.28.6+k3s2

networking:
  ssh:
    port: 22
    use_agent: false # set to true if your key has a passphrase
    public_key_path: "~/datacloud.pub"
    private_key_path: "~/datacloud"
  allowed_networks:
    ssh:
      - 0.0.0.0/0
    api:
      - 0.0.0.0/0
  public_network:
    ipv4: true
    ipv6: true
  private_network:
    enabled : true
    subnet: 10.0.0.0/16
    existing_network_name: ""
  cni:
    enabled: true
    encryption: false
    mode: flannel

manifests:
cloud_controller_manager_manifest_url: "https://github.com/hetznercloud/hcloud-cloud-controller-manager/releases/download/v1.20.0/ccm-networks.yaml"
csi_driver_manifest_url: "https://raw.githubusercontent.com/hetznercloud/csi-driver/v2.9.0/deploy/kubernetes/hcloud-csi.yml"
system_upgrade_controller_deployment_manifest_url: "https://github.com/rancher/system-upgrade-controller/releases/download/v0.13.4/system-upgrade-controller.yaml"
system_upgrade_controller_crd_manifest_url: "https://github.com/rancher/system-upgrade-controller/releases/download/v0.13.4/crd.yaml"
cluster_autoscaler_manifest_url: "https://raw.githubusercontent.com/kubernetes/autoscaler/master/cluster-autoscaler/cloudprovider/hetzner/examples/cluster-autoscaler-run-on-master.yaml"

datastore:
  mode: etcd # etcd (default) or external
  external_datastore_endpoint: postgres://....

schedule_workloads_on_masters: false

set include_instance_type_in_instance_name: true

masters_pool:
  instance_type: cx22
  location: nbg1
  instance_count: 3

worker_node_pools:
- name: nbg1-cx21
  instance_type: cx42
  instance_count: 4
  location: nbg1
  labels:
    - key: trino
      value: on

- name: nbg1-jupyterspark
  instance_type: cpx41
  instance_count: 1
  location: nbg1
  labels:
    - key: release
      value: jupyter

- name: nbg1-cpx41
  instance_type: cpx41
  instance_count: 0
  location: nbg1
  autoscaling:
    enabled: true
    min_instances: 0
    max_instances: 5

embedded_registry_mirror:
  enabled: true

post_create_commands:
- apt update
- apt upgrade -y

</p>
</details>

**Reply by vitobotta** [reply]
Cool, glad to hear :) Just keep in mind for now not to change the instance type in the config file for any pools until the upcoming release is available. With that one the instance type will only apply to new instances without affecting old ones.
